src/services/pcn-router/src/Ports.cpp

```diff
@@ -89,9 +89,69 @@ Ports::Ports(polycube::service::Cube<Ports> &parent,
     PortsSecondaryip::createInControlPlane(*this, addr.getIp(),
                                            addr.getNetmask(), addr);
   }
+
+  // lambda function to align IP and Netmask between Port and ExtIface
+  paremeter_event_callback f_ip;
+  f_ip = [&](const std::string name_iface, const std::string info_ipAddress) {
+    if (Ports::peer() == name_iface) {
```

Comment:
Any specific reason you are using the `Ports::` prefix, if not please remove it.